From d5acc989f4913beaa74d2af3f15a7f83848b59d4 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Tue, 26 Jun 2018 11:17:32 +0800
Subject: [PATCH] DRVLIB_SW-799 - Force SLC type in ltq_spinand

Kernel 4.9.109 prohibit the use of UBI for MLC, which is determined
from bits_per_cell property that ltq_spinand driver never set.
There are no unified way to determine SLC/MLC from flash. Some use
ONFI, others do not conform to this. So for now we simply force all
as SLC.
---
 drivers/mtd/ltq-spinand/ltq_spinand.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/mtd/ltq-spinand/ltq_spinand.c b/drivers/mtd/ltq-spinand/ltq_spinand.c
index a1bc833da319..0a623834ca4f 100644
--- a/drivers/mtd/ltq-spinand/ltq_spinand.c
+++ b/drivers/mtd/ltq-spinand/ltq_spinand.c
@@ -322,6 +322,12 @@ struct nand_flash_dev * spinand_flash_detect(struct mtd_info *mtd, struct nand_c
 
 	chip->chipsize = (type->chipsize) << 20;
 
+	/* There are no unified way to determine SLC/MLC from flash. Some use
+	 * ONFI, others do not conform to this. So for now we simply force all
+	 * as SLC.
+	 */
+	chip->bits_per_cell = 1;
+
 	/* do we need it ? */
 	chip->options |= type->options;
 
